test/spectests/spectests.cpp

```diff
@@ -256,6 +256,49 @@ class test_runner
                 fail("Invalid module parsed successfully. Expected error: " +
                      cmd.at("text").get<std::string>());
             }
+            else if (type == "assert_unlinkable" || type == "assert_uninstantiable")
+            {
+                // NOTE: assert_uninstantiable should result in a start function trap
+                //       assert_unlinkable checks all other instantiation failures
+
+                const auto module_type = cmd.at("module_type").get<std::string>();
+                if (module_type != "binary")
+                {
+                    skip("Only binary modules are supported.");
+                    continue;
+                }
+
+                const auto filename = cmd.at("filename").get<std::string>();
+                const auto wasm_binary = load_wasm_file(path, filename);
+                try
+                {
+                    fizzy::Module module = fizzy::parse(wasm_binary);
+
+                    auto imports = create_imports(module);
+                    if (!imports.has_value())
+                    {
+                        pass();
+                        continue;
+                    }
+
+                    fizzy::instantiate(std::move(module),
+                        std::move(imports->functions), std::move(imports->tables),
+                        std::move(imports->memories), std::move(imports->globals));
+                }
+                catch (const fizzy::parser_error& ex)
+                {
+                    fail(std::string{"Parsing failed with error: "} + ex.what());
+                    continue;
+                }
+                catch (const fizzy::instantiate_error&)
+                {
+                    pass();
```

Comment:
Should it check here for `instantiate` message ?

Reply:
Yes we can check.